tests/text_extraction_tests.py

```diff
@@ -1,98 +1,269 @@
+    def test_class_instantiation_with_no_filepath(self):
+        gazette = Gazette(
+            apache_tika_jar=self.TIKA_PATH,
+            content="tests/data/fake_gazette.txt"
         )
-        self.assertIsNotNone(text_file)
-        self.assertNotEqual(0, len(text_file))
-        self.check_if_text_has_the_fake_text(text_file)
+        self.assertEqual(gazette.filepath, None)
+        self.assertNotEqual(gazette.tika_jar, None)
+        self.assertNotEqual(gazette.content, None)
 
-    def test_extract_from_txt_file_should_return_text_file(self):
-        text = get_text_from_file(
-            "tests/data/fake_gazette.txt", apache_tika_jar=self.APACHE_TIKA_JAR_PATH
+    def test_class_instantiation_with_all_arguments(self):
+        gazette = Gazette(
+            filepath="tests/data/fake_gazette.pdf",
+            apache_tika_jar=self.TIKA_PATH,
+            content="tests/data/fake_gazette.txt"
         )
-        self.assertIsNotNone(text)
-        self.assertNotEqual(0, len(text))
-        self.check_if_text_has_the_fake_text(text)
-
-    def check_if_text_has_the_fake_text(self, text_file_path):
-        with open(text_file_path, "r") as f:
-            text = f.read()
-            self.assertIn(
-                "Hi this is a document created to test the text extraction for the Querido Diário project.",
-                text,
-                msg="Extracted text does not have the expected string",
-            )
+        self.assertNotEqual(gazette.filepath, None)
+        self.assertNotEqual(gazette.tika_jar, None)
+        self.assertNotEqual(gazette.content, None)
+
+
+    def test_extract_text_from_doc_should_return_content(self):
+        gazette = Gazette("tests/data/fake_gazette.doc", self.TIKA_PATH)
+
+        gazette.extract_content()
+        self.assertEqual(gazette.filepath, "tests/data/fake_gazette.txt")
+
+        gazette.load_content()
+        self.assertNotEqual(0, len(gazette.content))
+        self.assertIn("Querido", gazette.content, "Extraction Failed")
+
+    def test_extract_text_from_docx_should_return_content(self):
+        gazette = Gazette("tests/data/fake_gazette.docx", self.TIKA_PATH)
+
+        gazette.extract_content()
+        self.assertEqual(gazette.filepath, "tests/data/fake_gazette.txt")
+
+        gazette.load_content()
+        self.assertNotEqual(0, len(gazette.content))
+        self.assertIn("Querido", gazette.content, "Extraction Failed")
+
+    def test_extract_text_from_odt_should_return_content(self):
+        gazette = Gazette("tests/data/fake_gazette.odt", self.TIKA_PATH)
+
+        gazette.extract_content()
+        self.assertEqual(gazette.filepath, "tests/data/fake_gazette.txt")
+
+        gazette.load_content()
+        self.assertNotEqual(0, len(gazette.content))
+        self.assertIn("Querido", gazette.content, "Extraction Failed")
+
+    def test_extract_text_from_html_should_return_content(self):
+        gazette = Gazette("tests/data/fake_gazette.html", self.TIKA_PATH)
```

Comment:
This file is missing in the `tests/data` directory.

Reply:
Also a mistake... I'm now uploading all test files!